pkg/controllers/edgeconnect/controller_test.go

```diff
@@ -219,6 +222,111 @@ func TestReconcile(t *testing.T) {
+		ec := &edgeconnect.EdgeConnect{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      testName,
+				Namespace: testNamespace,
+			},
+			Spec: edgeconnect.EdgeConnectSpec{
+				ApiServer: "abc12345.dynatrace.com",
+				OAuth: edgeconnect.OAuthSpec{
+					Endpoint:     "https://test.com/sso/oauth2/token",
+					Resource:     "urn:dtenvironment:test12345",
+					ClientSecret: testOauthClientSecret,
+					Provisioner:  false,
+				},
+			},
+		}
+		controller := createFakeClientAndReconciler(t, ec,
+			createClientSecret(testOauthClientSecret, ec.Namespace),
+			createKubeSystemNamespace(),
+		)
+
+		_, err := controller.Reconcile(context.Background(), reconcile.Request{
+			NamespacedName: types.NamespacedName{Namespace: testNamespace, Name: testName},
+		})
+		require.NoError(t, err)
+
+		err = controller.apiReader.Get(context.TODO(), client.ObjectKey{Name: ec.Name, Namespace: ec.Namespace}, ec)
+		require.NoError(t, err)
+		require.NotEmpty(t, ec.Conditions())
+
+		condition := meta.FindStatusCondition(*ec.Conditions(), consts.SecretConfigConditionType)
+		assert.Equal(t, metav1.ConditionTrue, condition.Status)
+		assert.Equal(t, conditions.SecretCreatedReason, condition.Reason)
+		assert.Equal(t, ec.Name+"-"+consts.EdgeConnectSecretSuffix+" created", condition.Message)
+	})
+
+	t.Run(`SecretConfigConditionType is set SecretGenFailed failed to get clientSecret`, func(t *testing.T) {
+		ec := &edgeconnect.EdgeConnect{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      testName,
+				Namespace: testNamespace,
+			},
+			Spec: edgeconnect.EdgeConnectSpec{
+				ApiServer: "abc12345.dynatrace.com",
+				OAuth: edgeconnect.OAuthSpec{
+					Endpoint:     "https://test.com/sso/oauth2/token",
+					Resource:     "urn:dtenvironment:test12345",
+					ClientSecret: testOauthClientSecret,
+					Provisioner:  false,
+				},
+			},
+		}
+
+		controller := createFakeClientAndReconciler(t, ec,
+			createKubeSystemNamespace(),
+		)
+
+		err := controller.reconcileEdgeConnectRegular(context.Background(), ec)
+		require.Error(t, err)
+		require.NotEmpty(t, ec.Conditions())
+
+		condition := meta.FindStatusCondition(*ec.Conditions(), consts.SecretConfigConditionType)
+		assert.Equal(t, metav1.ConditionFalse, condition.Status)
+		assert.Equal(t, conditions.SecretGenerationFailed, condition.Reason)
+		assert.Contains(t, condition.Message, "Failed to generate secret: failed to get clientSecret")
+	})
+
+	t.Run(`SecretConfigConditionType is set SecretGenFailed failed`, func(t *testing.T) {
+		ec := &edgeconnect.EdgeConnect{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      testName,
+				Namespace: testNamespace,
+			},
+			Spec: edgeconnect.EdgeConnectSpec{
+				ApiServer: "abc12345.dynatrace.com",
+				OAuth: edgeconnect.OAuthSpec{
+					Endpoint:     "https://test.com/sso/oauth2/token",
+					Resource:     "urn:dtenvironment:test12345",
+					ClientSecret: testOauthClientSecret,
+					Provisioner:  false,
+				},
+			},
+		}
```

Comment:
We could move this to a function to not repeat it in the code, WDYT?

Reply:
Repeating in tests is okey. Anyway we need big refactoring of EC. So we will refactor tests as well.